service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,117 @@
+            )
+            .build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void shouldReturnHearingDateDynamicListWhenCmoHasPreviousSelectedValue() {
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date),
+                "caseManagementOrder", createCaseManagementOrder(draftCMOService.convertDate(date.plusDays(2)))
+                )).build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void shouldReturnHearingDateDynamicListWhenHearingDatesNotNull() {
+        DynamicList hearingList = draftCMOService.buildDynamicListFromHearingDetails(
+            createHearingBookings(date));
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void convertDateShouldReturnDateInUKFormat() {
+        String dateInUKFormat = draftCMOService.convertDate(LocalDate.of(2019,11,11));
+        assertThat(dateInUKFormat).isEqualTo("11 Nov 2019");
+    }
+
+    private CaseManagementOrder createCaseManagementOrder(String hearingDate) {
+        return CaseManagementOrder.builder()
+            .hearingDate(hearingDate)
+            .hearingDateId(UUID.fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2").toString()).build();
+    }
+
+    private List<Element<HearingBooking>> createHearingBookings(LocalDate now) {
+        return ImmutableList.of(
+            Element.<HearingBooking>builder()
+                .id(UUID.randomUUID())
+                .value(createHearingBooking(now.plusDays(5)))
+                .build(),
+            Element.<HearingBooking>builder()
+                .id(UUID.fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2"))
```

Comment:
```suggestion
                .id("6b3ee98f-acff-4b64-bb00-cc3db02a24b2")
```

Reply:
id needs UUID parameter. Suggested change is given error. model is like ```public class Element<T> {
    private final UUID id;
    @NotNull
    @Valid
    private final T value;
}```

Reply:
Set randomUUID() to a variable then 👍